### PR TITLE
Remove duplicate and unnecessary use statements

### PR DESCRIPTION
**Pull request by rmontuoro**
## Description

This PR removes duplicate and unnecessary `use` statements.

## Motivation and Context

Code maintenance and cleanup
## How Has This Been Tested?

These changes were tested on Orion, building MAPL as standalone library and using it to build and run UFS-GOCART.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Trivial change (affects only documentation or cleanup)

## Checklist:

- [ ] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [ ] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

**Review by tclune — APPROVED** [review]

File: generic/MaplGeneric.F90 [hunk 1]
```diff
@@ -2,7 +2,6 @@ module MaplGeneric
    use mapl_AbstractComponent
    use mapl_MaplComponent
    use mapl_ComponentSpecification
-   use mapl_VariableSpecification
```
Comment (tclune) [inline generic/MaplGeneric.F90:5]: I still don't think this change is required.   And I'll likely need to put it right back as soon as I continue refactoring MAPL.

**Review by tclune — CHANGES_REQUESTED** [review]
See inline comment.

File: generic/MaplGeneric.F90 (hunk 1 above)
Comment (rmontuoro) [reply generic/MaplGeneric.F90]: UFS-GOCART builds correctly after reintroducing this use statement  Thanks, @tclune, for catching this unnecessary change. Apologies for the mix-up.

**Review by tclune — DISMISSED** [review]

**Comment by mathomp4** [comment]
@rmontuoro I believe this PR breaks GEOS. When I tried to build your PRs with GEOS I got:
```
[ 97%] Linking Fortran executable ../../../../../../bin/StandAlone_AdvCore.x
ld: ../../../../../../lib/libFVdycoreCubed_GridComp.a(CubeToLatLonRegridder.F90.o): in function `cubetolatlonregriddermod_mp_initialize_subclass_':
/discover/swdev/mathomp4/Models/GEOSgcm-RafaelleMAPL/GEOSgcm/src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSsuperdyn_GridComp/@FVdycoreCubed_GridComp/CubeToLatLonRegridder.F90:90: undefined reference to `mapl_grid_interior_'
ld: ../../../../../../lib/libFVdycoreCubed_GridComp.a(LatLonToCubeRegridder.F90.o): in function `latlontocuberegriddermod_mp_initialize_subclass_':
/discover/swdev/mathomp4/Models/GEOSgcm-RafaelleMAPL/GEOSgcm/src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSsuperdyn_GridComp/@FVdycoreCubed_GridComp/LatLonToCubeRegridder.F90:86: undefined reference to `mapl_grid_interior_'
src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSsuperdyn_GridComp/@FVdycoreCubed_GridComp/CMakeFiles/StandAlone_AdvCore.x.dir/build.make:132: recipe for target 'bin/StandAlone_AdvCore.x' failed
```

**Comment by mathomp4** [comment]
These are the same errors CI is reporting:
```
/usr/bin/ld: ../../../../../../lib/libFVdycoreCubed_GridComp.a(CubeToLatLonRegridder.F90.o): in function `__cubetolatlonregriddermod_MOD_initialize_subclass':
/root/project/GEOSgcm/src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSsuperdyn_GridComp/@FVdycoreCubed_GridComp/CubeToLatLonRegridder.F90:90: undefined reference to `mapl_grid_interior_'
/usr/bin/ld: ../../../../../../lib/libFVdycoreCubed_GridComp.a(LatLonToCubeRegridder.F90.o): in function `__latlontocuberegriddermod_MOD_initialize_subclass':
/root/project/GEOSgcm/src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSsuperdyn_GridComp/@FVdycoreCubed_GridComp/LatLonToCubeRegridder.F90:86: undefined reference to `mapl_grid_interior_'
```

**Comment by rmontuoro** [comment]
@mathomp4 - It seems the rename clause from the second `use MAPL_BaseMod` statement overrides the visibility of `MAPL_GRID_INTERIOR` from the first `use MAPL_BaseMod`[statement](https://github.com/rmontuoro/MAPL/blob/567e141176a42cbd0cfba8103f7231b6baa04d0a/base/MAPL_Mod.F90#L7-L9):
```
  use MAPL_BaseMod
! For temporary backward compatibility after moving/renaming:
  use MAPL_BaseMod, only: ESMF_GRID_INTERIOR => MAPL_GRID_INTERIOR
```

Commit b30690d restores visibility of `MAPL_GRID_INTERIOR`.

**Comment by mathomp4** [comment]
@rmontuoro Thanks. You might also need to merge in our `develop` and push on your two PRs. GitHub isn't letting us merge them in here and we don't seem to have the "Update Branch" button. Or, well, I don't. I wonder if @WilliamJamieson would have rights to?

**Comment by rmontuoro** [comment]
@mathomp4 - I've updated both PR branches to the latest `develop` revision.

**Review by WilliamJamieson — APPROVED** [review]

**Review by tclune — APPROVED** [review]

**Comment by WilliamJamieson** [comment]
I ran GEOS with the changes from this PR and PR #721 and found that the results where zero-diff with the results from using MAPL develop.

**Pull request merged**